### ISSUE TITLE
pmml exported by python can't be analysed correctly

### ISSUE THREAD

**Issue by ftutor** [jpmml/jpmml-evaluator#74]
recently, I exported a GBDT pmml  using sklearn2pmml, howerer when I used jpmml-evaluator to deploy this pmml, I found that the outputs  were different between java output and python output with the same inputs . for example.
inputs:"3420","0","107.2916667","107.2916667","2575","3823","3420","0","533","0","2575","2575","1665","2","2575","1665","2"

results:
python:10756.44321344
java:10564.699630156896

less than 0.2% difference.

the model has attached

**Comment by ftutor** [comment]
it seems file can't be attached ,I tried several times and failed
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<PMML xmlns="http://www.dmg.org/PMML-4_3" version="4.3">
	<Header>
		<Application name="JPMML-SkLearn" version="1.3.7"/>
		<Timestamp>2017-09-09T06:51:13Z</Timestamp>
	</Header>
	<MiningBuildTask>
		<Extension>PMMLPipeline(steps=[('regressor', GradientBoostingRegressor(alpha=0.9, init=None, learning_rate=0.01, loss='ls',
             max_depth=3, max_features='sqrt', max_leaf_nodes=None,
             min_samples_leaf=20, min_samples_split=2,
             min_weight_fraction_leaf=0.0, n_estimators=10, random_state=0,
             subsample=1.0, verbose=0, warm_start=False))])</Extension>
	</MiningBuildTask>
	<DataDictionary>
		<DataField name="y" optype="continuous" dataType="double"/>
		<DataField name="x1" optype="continuous" dataType="float"/>
		<DataField name="x2" optype="continuous" dataType="float"/>
		<DataField name="x3" optype="continuous" dataType="float"/>
		<DataField name="x4" optype="continuous" dataType="float"/>
		<DataField name="x5" optype="continuous" dataType="float"/>
		<DataField name="x6" optype="continuous" dataType="float"/>
		<DataField name="x7" optype="continuous" dataType="float"/>
		<DataField name="x8" optype="continuous" dataType="float"/>
		<DataField name="x9" optype="continuous" dataType="float"/>
		<DataField name="x10" optype="continuous" dataType="float"/>
		<DataField name="x11" optype="continuous" dataType="float"/>
		<DataField name="x12" optype="continuous" dataType="float"/>
		<DataField name="x13" optype="continuous" dataType="float"/>
		<DataField name="x15" optype="continuous" dataType="float"/>
		<DataField name="x16" optype="continuous" dataType="float"/>
	</DataDictionary>
	<MiningModel functionName="regression">
		<MiningSchema>
			<MiningField name="y" usageType="target"/>
			<MiningField name="x3"/>
			<MiningField name="x7"/>
			<MiningField name="x8"/>
			<MiningField name="x6"/>
			<MiningField name="x13"/>
			<MiningField name="x2"/>
			<MiningField name="x5"/>
			<MiningField name="x10"/>
			<MiningField name="x1"/>
			<MiningField name="x16"/>
			<MiningField name="x9"/>
			<MiningField name="x12"/>
			<MiningField name="x15"/>
			<MiningField name="x11"/>
			<MiningField name="x4"/>
		</MiningSchema>
		<Targets>
			<Target field="y" rescaleConstant="10688.717982571956" rescaleFactor="0.01"/>
		</Targets>
		<Segmentation multipleModelMethod="sum">
			<Segment id="1">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x1"/>
						<MiningField name="x2"/>
						<MiningField name="x3"/>
						<MiningField name="x6"/>
						<MiningField name="x9"/>
						<MiningField name="x11"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x3" operator="lessOrEqual" value="241.375"/>
							<Node id="3">
								<SimplePredicate field="x9" operator="lessOrEqual" value="459.5"/>
								<Node id="4" score="-3086.727665196687">
									<SimplePredicate field="x6" operator="lessOrEqual" value="2887"/>
								</Node>
								<Node id="5" score="-964.3017737807791">
									<SimplePredicate field="x6" operator="greaterThan" value="2887"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x9" operator="greaterThan" value="459.5"/>
								<Node id="7" score="-608.4274031071363">
									<SimplePredicate field="x2" operator="lessOrEqual" value="8837"/>
								</Node>
								<Node id="8" score="1820.2632506719608">
									<SimplePredicate field="x2" operator="greaterThan" value="8837"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x3" operator="greaterThan" value="241.375"/>
							<Node id="10">
								<SimplePredicate field="x1" operator="lessOrEqual" value="2522"/>
								<Node id="11" score="-455.4877644020123">
									<SimplePredicate field="x11" operator="lessOrEqual" value="15044.5"/>
								</Node>
								<Node id="12" score="2107.1062276251964">
									<SimplePredicate field="x11" operator="greaterThan" value="15044.5"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x1" operator="greaterThan" value="2522"/>
								<Node id="14" score="3325.3629955802717">
									<SimplePredicate field="x6" operator="lessOrEqual" value="35299.5"/>
								</Node>
								<Node id="15" score="5451.158898413221">
									<SimplePredicate field="x6" operator="greaterThan" value="35299.5"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="2">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x1"/>
						<MiningField name="x5"/>
						<MiningField name="x7"/>
						<MiningField name="x10"/>
						<MiningField name="x12"/>
						<MiningField name="x13"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x13" operator="lessOrEqual" value="2579"/>
							<Node id="3">
								<SimplePredicate field="x1" operator="lessOrEqual" value="2489.5"/>
								<Node id="4" score="-3402.636179686948">
									<SimplePredicate field="x7" operator="lessOrEqual" value="889.5"/>
								</Node>
								<Node id="5" score="-1533.9470213022887">
									<SimplePredicate field="x7" operator="greaterThan" value="889.5"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x1" operator="greaterThan" value="2489.5"/>
								<Node id="7" score="534.2185487576107">
									<SimplePredicate field="x12" operator="lessOrEqual" value="732.5"/>
								</Node>
								<Node id="8" score="2178.8027039432936">
									<SimplePredicate field="x12" operator="greaterThan" value="732.5"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x13" operator="greaterThan" value="2579"/>
							<Node id="10">
								<SimplePredicate field="x1" operator="lessOrEqual" value="2522"/>
								<Node id="11" score="-802.8994769778891">
									<SimplePredicate field="x5" operator="lessOrEqual" value="15044.5"/>
								</Node>
								<Node id="12" score="2087.1887190101593">
									<SimplePredicate field="x5" operator="greaterThan" value="15044.5"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x1" operator="greaterThan" value="2522"/>
								<Node id="14" score="2659.3953543619136">
									<SimplePredicate field="x10" operator="lessOrEqual" value="2635.5"/>
								</Node>
								<Node id="15" score="4882.106002086271">
									<SimplePredicate field="x10" operator="greaterThan" value="2635.5"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="3">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x2"/>
						<MiningField name="x4"/>
						<MiningField name="x8"/>
						<MiningField name="x9"/>
						<MiningField name="x11"/>
						<MiningField name="x12"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x11" operator="lessOrEqual" value="5793"/>
							<Node id="3">
								<SimplePredicate field="x2" operator="lessOrEqual" value="2305.5"/>
								<Node id="4" score="-2658.438002863537">
									<SimplePredicate field="x12" operator="lessOrEqual" value="779"/>
								</Node>
								<Node id="5" score="-884.4244893198668">
									<SimplePredicate field="x12" operator="greaterThan" value="779"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x2" operator="greaterThan" value="2305.5"/>
								<Node id="7" score="-918.2568045684706">
									<SimplePredicate field="x9" operator="lessOrEqual" value="489"/>
								</Node>
								<Node id="8" score="705.7934420011668">
									<SimplePredicate field="x9" operator="greaterThan" value="489"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x11" operator="greaterThan" value="5793"/>
							<Node id="10">
								<SimplePredicate field="x8" operator="lessOrEqual" value="2.5"/>
								<Node id="11" score="-734.4868755781721">
									<SimplePredicate field="x9" operator="lessOrEqual" value="483.5"/>
								</Node>
								<Node id="12" score="1934.129500131445">
									<SimplePredicate field="x9" operator="greaterThan" value="483.5"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x8" operator="greaterThan" value="2.5"/>
								<Node id="14" score="3537.1138503369525">
									<SimplePredicate field="x4" operator="lessOrEqual" value="1463.125"/>
								</Node>
								<Node id="15" score="5599.243638517996">
									<SimplePredicate field="x4" operator="greaterThan" value="1463.125"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="4">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x7"/>
						<MiningField name="x9"/>
						<MiningField name="x10"/>
						<MiningField name="x11"/>
						<MiningField name="x13"/>
						<MiningField name="x15"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x11" operator="lessOrEqual" value="5793"/>
							<Node id="3">
								<SimplePredicate field="x7" operator="lessOrEqual" value="2430.5"/>
								<Node id="4" score="-2743.1905535903134">
									<SimplePredicate field="x10" operator="lessOrEqual" value="284.5"/>
								</Node>
								<Node id="5" score="-465.3407980752651">
									<SimplePredicate field="x10" operator="greaterThan" value="284.5"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x7" operator="greaterThan" value="2430.5"/>
								<Node id="7" score="424.8827241871386">
									<SimplePredicate field="x13" operator="lessOrEqual" value="467"/>
								</Node>
								<Node id="8" score="1809.5970554748542">
									<SimplePredicate field="x13" operator="greaterThan" value="467"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x11" operator="greaterThan" value="5793"/>
							<Node id="10">
								<SimplePredicate field="x11" operator="lessOrEqual" value="32515"/>
								<Node id="11" score="-688.1655400553246">
									<SimplePredicate field="x9" operator="lessOrEqual" value="479.5"/>
								</Node>
								<Node id="12" score="2203.836924544765">
									<SimplePredicate field="x9" operator="greaterThan" value="479.5"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x11" operator="greaterThan" value="32515"/>
								<Node id="14" score="3894.22182892508">
									<SimplePredicate field="x15" operator="lessOrEqual" value="165994.25"/>
								</Node>
								<Node id="15" score="6096.065494999401">
									<SimplePredicate field="x15" operator="greaterThan" value="165994.25"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="5">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x1"/>
						<MiningField name="x2"/>
						<MiningField name="x5"/>
						<MiningField name="x7"/>
						<MiningField name="x10"/>
						<MiningField name="x11"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x11" operator="lessOrEqual" value="5793"/>
							<Node id="3">
								<SimplePredicate field="x1" operator="lessOrEqual" value="1833.5"/>
								<Node id="4" score="-3618.9803461187307">
									<SimplePredicate field="x2" operator="lessOrEqual" value="442.5"/>
								</Node>
								<Node id="5" score="-1828.5243850070865">
									<SimplePredicate field="x2" operator="greaterThan" value="442.5"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x1" operator="greaterThan" value="1833.5"/>
								<Node id="7" score="288.3701616792764">
									<SimplePredicate field="x10" operator="lessOrEqual" value="4470.5"/>
								</Node>
								<Node id="8" score="2492.154906695982">
									<SimplePredicate field="x10" operator="greaterThan" value="4470.5"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x11" operator="greaterThan" value="5793"/>
							<Node id="10">
								<SimplePredicate field="x7" operator="lessOrEqual" value="2714.5"/>
								<Node id="11" score="575.6037103779778">
									<SimplePredicate field="x5" operator="lessOrEqual" value="29631"/>
								</Node>
								<Node id="12" score="3641.65120571965">
									<SimplePredicate field="x5" operator="greaterThan" value="29631"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x7" operator="greaterThan" value="2714.5"/>
								<Node id="14" score="2752.0795267862873">
									<SimplePredicate field="x2" operator="lessOrEqual" value="8233"/>
								</Node>
								<Node id="15" score="4878.404497657828">
									<SimplePredicate field="x2" operator="greaterThan" value="8233"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="6">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x1"/>
						<MiningField name="x3"/>
						<MiningField name="x8"/>
						<MiningField name="x9"/>
						<MiningField name="x10"/>
						<MiningField name="x13"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x3" operator="lessOrEqual" value="241.375"/>
							<Node id="3">
								<SimplePredicate field="x9" operator="lessOrEqual" value="459.5"/>
								<Node id="4" score="-3056.474817158997">
									<SimplePredicate field="x1" operator="lessOrEqual" value="2496"/>
								</Node>
								<Node id="5" score="135.98042887552668">
									<SimplePredicate field="x1" operator="greaterThan" value="2496"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x9" operator="greaterThan" value="459.5"/>
								<Node id="7" score="-901.7261959698495">
									<SimplePredicate field="x13" operator="lessOrEqual" value="619"/>
								</Node>
								<Node id="8" score="490.36266302394716">
									<SimplePredicate field="x13" operator="greaterThan" value="619"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x3" operator="greaterThan" value="241.375"/>
							<Node id="10">
								<SimplePredicate field="x1" operator="lessOrEqual" value="2522"/>
								<Node id="11" score="3.735870401984938">
									<SimplePredicate field="x8" operator="lessOrEqual" value="2.5"/>
								</Node>
								<Node id="12" score="2941.7556283177455">
									<SimplePredicate field="x8" operator="greaterThan" value="2.5"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x1" operator="greaterThan" value="2522"/>
								<Node id="14" score="2732.520800950917">
									<SimplePredicate field="x10" operator="lessOrEqual" value="2635.5"/>
								</Node>
								<Node id="15" score="4933.151964430475">
									<SimplePredicate field="x10" operator="greaterThan" value="2635.5"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="7">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x5"/>
						<MiningField name="x6"/>
						<MiningField name="x8"/>
						<MiningField name="x9"/>
						<MiningField name="x10"/>
						<MiningField name="x11"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x6" operator="lessOrEqual" value="5600.5"/>
							<Node id="3">
								<SimplePredicate field="x10" operator="lessOrEqual" value="284.5"/>
								<Node id="4" score="-2474.132611309258">
									<SimplePredicate field="x11" operator="lessOrEqual" value="721.5"/>
								</Node>
								<Node id="5" score="-894.0915041601693">
									<SimplePredicate field="x11" operator="greaterThan" value="721.5"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x10" operator="greaterThan" value="284.5"/>
								<Node id="7" score="-1203.7890253039045">
									<SimplePredicate field="x9" operator="lessOrEqual" value="489"/>
								</Node>
								<Node id="8" score="666.8254682547501">
									<SimplePredicate field="x9" operator="greaterThan" value="489"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x6" operator="greaterThan" value="5600.5"/>
							<Node id="10">
								<SimplePredicate field="x8" operator="lessOrEqual" value="2.5"/>
								<Node id="11" score="371.74223629823047">
									<SimplePredicate field="x5" operator="lessOrEqual" value="15496.5"/>
								</Node>
								<Node id="12" score="2341.581563332378">
									<SimplePredicate field="x5" operator="greaterThan" value="15496.5"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x8" operator="greaterThan" value="2.5"/>
								<Node id="14" score="3358.8733798765215">
									<SimplePredicate field="x5" operator="lessOrEqual" value="35115"/>
								</Node>
								<Node id="15" score="5415.702071263784">
									<SimplePredicate field="x5" operator="greaterThan" value="35115"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="8">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x1"/>
						<MiningField name="x6"/>
						<MiningField name="x10"/>
						<MiningField name="x11"/>
						<MiningField name="x16"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x6" operator="lessOrEqual" value="5600.5"/>
							<Node id="3">
								<SimplePredicate field="x1" operator="lessOrEqual" value="1833.5"/>
								<Node id="4" score="-3080.582867112955">
									<SimplePredicate field="x16" operator="lessOrEqual" value="856.5"/>
								</Node>
								<Node id="5" score="-1852.9781324505211">
									<SimplePredicate field="x16" operator="greaterThan" value="856.5"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x1" operator="greaterThan" value="1833.5"/>
								<Node id="7" score="-280.5635884906094">
									<SimplePredicate field="x1" operator="lessOrEqual" value="3270.5"/>
								</Node>
								<Node id="8" score="2087.860250198625">
									<SimplePredicate field="x1" operator="greaterThan" value="3270.5"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x6" operator="greaterThan" value="5600.5"/>
							<Node id="10">
								<SimplePredicate field="x1" operator="lessOrEqual" value="2522"/>
								<Node id="11" score="-129.04216770469057">
									<SimplePredicate field="x10" operator="lessOrEqual" value="1855.5"/>
								</Node>
								<Node id="12" score="2521.1177176287365">
									<SimplePredicate field="x10" operator="greaterThan" value="1855.5"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x1" operator="greaterThan" value="2522"/>
								<Node id="14" score="2816.367897311233">
									<SimplePredicate field="x11" operator="lessOrEqual" value="24016"/>
								</Node>
								<Node id="15" score="4830.010418628921">
									<SimplePredicate field="x11" operator="greaterThan" value="24016"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="9">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x2"/>
						<MiningField name="x5"/>
						<MiningField name="x6"/>
						<MiningField name="x7"/>
						<MiningField name="x8"/>
						<MiningField name="x9"/>
						<MiningField name="x13"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x6" operator="lessOrEqual" value="5600.5"/>
							<Node id="3">
								<SimplePredicate field="x7" operator="lessOrEqual" value="2489.5"/>
								<Node id="4" score="-3131.095714750387">
									<SimplePredicate field="x2" operator="lessOrEqual" value="426.5"/>
								</Node>
								<Node id="5" score="-1207.1600712353152">
									<SimplePredicate field="x2" operator="greaterThan" value="426.5"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x7" operator="greaterThan" value="2489.5"/>
								<Node id="7" score="508.5735179875147">
									<SimplePredicate field="x13" operator="lessOrEqual" value="467"/>
								</Node>
								<Node id="8" score="1815.9824318169572">
									<SimplePredicate field="x13" operator="greaterThan" value="467"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x6" operator="greaterThan" value="5600.5"/>
							<Node id="10">
								<SimplePredicate field="x8" operator="lessOrEqual" value="2.5"/>
								<Node id="11" score="-680.8904745298024">
									<SimplePredicate field="x9" operator="lessOrEqual" value="483.5"/>
								</Node>
								<Node id="12" score="1826.7255003411533">
									<SimplePredicate field="x9" operator="greaterThan" value="483.5"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x8" operator="greaterThan" value="2.5"/>
								<Node id="14" score="3296.021601260606">
									<SimplePredicate field="x5" operator="lessOrEqual" value="35115"/>
								</Node>
								<Node id="15" score="5319.9191202448255">
									<SimplePredicate field="x5" operator="greaterThan" value="35115"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
			<Segment id="10">
				<True/>
				<TreeModel functionName="regression" splitCharacteristic="binarySplit">
					<MiningSchema>
						<MiningField name="x3"/>
						<MiningField name="x4"/>
						<MiningField name="x5"/>
						<MiningField name="x9"/>
						<MiningField name="x10"/>
						<MiningField name="x16"/>
					</MiningSchema>
					<Node id="1">
						<True/>
						<Node id="2">
							<SimplePredicate field="x5" operator="lessOrEqual" value="5793"/>
							<Node id="3">
								<SimplePredicate field="x9" operator="lessOrEqual" value="459.5"/>
								<Node id="4" score="-2960.5170854154576">
									<SimplePredicate field="x10" operator="lessOrEqual" value="291.5"/>
								</Node>
								<Node id="5" score="-1015.3206979572631">
									<SimplePredicate field="x10" operator="greaterThan" value="291.5"/>
								</Node>
							</Node>
							<Node id="6">
								<SimplePredicate field="x9" operator="greaterThan" value="459.5"/>
								<Node id="7" score="-857.6698794810806">
									<SimplePredicate field="x16" operator="lessOrEqual" value="619"/>
								</Node>
								<Node id="8" score="488.49121703407366">
									<SimplePredicate field="x16" operator="greaterThan" value="619"/>
								</Node>
							</Node>
						</Node>
						<Node id="9">
							<SimplePredicate field="x5" operator="greaterThan" value="5793"/>
							<Node id="10">
								<SimplePredicate field="x10" operator="lessOrEqual" value="1773"/>
								<Node id="11" score="158.48049511235033">
									<SimplePredicate field="x4" operator="lessOrEqual" value="644.6458"/>
								</Node>
								<Node id="12" score="1947.6953458585099">
									<SimplePredicate field="x4" operator="greaterThan" value="644.6458"/>
								</Node>
							</Node>
							<Node id="13">
								<SimplePredicate field="x10" operator="greaterThan" value="1773"/>
								<Node id="14" score="2945.494105916916">
									<SimplePredicate field="x3" operator="lessOrEqual" value="1347.8334"/>
								</Node>
								<Node id="15" score="5148.222388154827">
									<SimplePredicate field="x3" operator="greaterThan" value="1347.8334"/>
								</Node>
							</Node>
						</Node>
					</Node>
				</TreeModel>
			</Segment>
		</Segmentation>
	</MiningModel>
</PMML>

**Comment by vruusmann** [comment]
Please attach the original GBDT pipeline in Pickle data format, plus a sample dataset in CSV data format (around ten data records, for which Python and (J)PMML give different results). Otherwise I'll just close this issue as "not enough evidence/not reproducible".

GitHub does not allow attachments in specific data formats. You can work around this limitation by changing the filename extension to `.txt`. For example, rename `supporting_info.zip` to `supporting_info.zip.txt` before attaching it.

**Comment by ftutor** [comment]
hi Vruusmann ,
Thanks for your quick reply.
I attached some files according to your requirement.
Python version:2.7
I am looking forward to your further reply.
Thx
[pmmltest.zip.txt](https://github.com/jpmml/jpmml-evaluator/files/1291204/pmmltest.zip.txt)

**Comment by ftutor** [comment]
I found the reason and fixed it.
`List<InputField> inputFields = evaluator.getInputFields();`

the sequence of **inputFields ** is different from the sequence of columns in the inputs ,which results in 
different outputs between java and python, I am wondering how we can align the  inputFields of evaluator with the location of variables in the input. currently, I established the relationship manually

**Comment by vruusmann** [comment]
> I am wondering how we can align the inputFields of evaluator with the location of variables in the input.

Scikit-Learn identifies columns by position: `X[0]`, `X[1]`, .., `X[n]`.
(J)PMML identifies columns by name: `X["first_column"]`, `X["second_column"]`, .., `X["last_column"]`.

The (J)PMML approach is much more robust, because it doesn't care if you reorder columns, add new columns or delete existing (but unused) columns.

The JPMML-Evaluator API enforces this practice by making the input and output interface of the  main evaluate method bound to the `java.util.Map<FieldName, Object>` type:
```java
Map<FieldName, ?> arguments = ...;
Map<FieldName, ?> results = evaluator.evaluate(arguments);
```

The ordering of key-value pairs in the above `arguments` and `results` map objects is completely irrelevant, as the values are retrieved and inserted by name:
```java
Map<FieldName, Object> arguments = new HashMap<>();
arguments.put(FieldName.create("second_columns"), Math.PI);
```

**Comment by vruusmann** [comment]
Does the above comment (https://github.com/jpmml/jpmml-evaluator/issues/74#issuecomment-328432534) mean that JPMML-Evaluator is giving you correct predictions now for the whole dataset?

**Comment by vruusmann** [comment]
Looking into your example ZIP file, then you are not taking the maximum out of the `sklearn2pmml` package in your Python script.

Namely, if your `X` dataset is an instance of `pandas.DataFrame`, and you fit the pipeline using `PMMLPipeline.fit(X, y)`, then the fitted pipeline will keep a list of column names as a `PMMLPipeline.active_fields` attribute. The JPMML-SkLearn library looks for this attribute, and uses this information to rename all fields in the PMML document appropriately.

In other words, if you refactored your Python script, then the resulting PMML document will contain field names such as `signup_first`, `amount_htl2y`, etc. (instead of current names `x1`, `x2`, etc.).

**Comment by ftutor** [comment]
hi Vruusmann，
At present，JPMML-Evaluator is giving me correct predictions for the whole dataset, and 
I already got field names such as `signup_first,amount_htl2y`etc instead of `x1,x2` etc.
I am very appreciated for your kindly support.

**Comment by ftutor** [comment]
hi Vruusmann，
At present，JPMML-Evaluator is giving me correct predictions for the whole dataset, and 
I already got field names such as `signup_first,amount_htl2y`etc instead of `x1,x2` etc.
I am very appreciated for your kindly support.

**Comment by ftutor** [comment]
hi Vruusmann，
At present，JPMML-Evaluator is giving me correct predictions for the whole dataset, and 
I already got field names such as `signup_first,amount_htl2y`etc instead of `x1,x2` etc.
I am very appreciated for your kindly support.

**Comment by ftutor** [comment]
As it is not an issue, you can close it.
